### PR TITLE
[#189] Convert all valid urls from http to https

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes part of #189 

- Did a project wide replacement of `http` -> `https`
- Ran [dead link checker](https://www.deadlinkchecker.com/website-dead-link-checker.asp) to scan for dead links
- From the broken links provided by dead link chapter, verify every broken link and convert back some of those dead links that are valid in `http` but not in `https`

The full log of dead links: 
[Dead Link Checker report - https___deploy-preview-190--learningresources.netlify.com_.pdf](https://github.com/se-edu/learningresources/files/4470611/Dead.Link.Checker.report.-.https___deploy-preview-190--learningresources.netlify.com_.pdf)

### CODE REVIEW

**Comment by jamessspanggg** [comment]
@damithc this PR won't solve the issue with mixed insecure content as some content is only available in `http`. But this PR changes all available links from `http` -> `https`

**Comment by jamessspanggg** [comment]
Will recheck if all the links changed are valid.

File: contents/node/nodejs.md [hunk 1]
```diff
@@ -181,14 +181,14 @@ Apart from acting as <tooltip content="See here for some common use cases of npm
...
+Node comes bundled with useful [core modules](https://nodejs.org/api/) such as the `fs` (file system) module which includes classes, methods and events to deal with file I/O operations and the `https` module which helps Node to transfer data over HTTP.
 There are also many useful and well-tested modules maintained by the community and external developers such as <tooltip content="A library providing various useful wrapper functions over JavaScript's promises">*Bluebird*</tooltip> on <tooltip content="the largest ecosystem of open source libraries in the world!">npm</tooltip>.
 
 This makes development a breeze -- if a specific functionality has a large development overhead, you could look to such modules to speed things up.
 
 Importing modules is easy - simply use the `require()` function and provide the module identifier or the file path.
 ```js
-const http = require('http'); // import a core module
+const https = require('https'); // import a core module
```
Comment (damithc) [inline contents/node/nodejs.md:191]: Is this correct?

**Review by damithc — COMMENTED** [review]
A query

File: contents/node/nodejs.md (hunk 1 above)
Comment (jamessspanggg) [reply contents/node/nodejs.md:191]: Opps, this is a mistake, but it is still a correct syntax as there is a `https` nodejs module that does the same thing. Will correct this

Comment (jamessspanggg) [reply contents/node/nodejs.md:191]: done.

**Comment by jamessspanggg** [comment]
Actually, there is a log showing the mixed contents: https://app.netlify.com/sites/learningresources/deploys/5e94a08b1b170d0007270f72

Screen Shot 2020-04-14 at 1 28 28 AM

Note that the links are not available in `https`


Log of mixed contents for the master branch: https://app.netlify.com/sites/learningresources/deploys/5e929bcda87b430006f27ab0

Screen Shot 2020-04-14 at 1 31 08 AM

**Comment by damithc** [comment]
> Actually, there is a log showing the mixed contents: https://app.netlify.com/sites/learningresources/deploys/5e94a08b1b170d0007270f72

Thanks for investigating. Can add this info to the relevant issue. #189

**Pull request merged**